build_system/ninja_utils.ml

```diff
@@ -0,0 +1,81 @@
+(* This file is part of the Catala build system, a specification language for tax and social
+   benefits computation rules. Copyright (C) 2020 Inria, contributor: Emile Rolley
+   <emile.rolley@tuta.io>
+
+   Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+   in compliance with the License. You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software distributed under the License
+   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+   or implied. See the License for the specific language governing permissions and limitations under
+   the License. *)
+
+module Expr = struct
+  type t = Seq of t list | Lit of string | Var of string
+
+  let rec to_string = function
+    | Lit s -> s
+    | Var s -> "$" ^ s
+    | Seq ls -> List.fold_left (fun acc s -> acc ^ " " ^ to_string s) "" ls
+
+  let list_to_string ?(sep = " ") ls = ls |> List.map to_string |> String.concat sep
+end
+
+module Rule = struct
+  type t = { name : string; command : Expr.t; description : Expr.t option }
+
+  let make name ~command ~description = { name; command; description = Option.some description }
+
+  let to_string rule =
+    Printf.sprintf "rule %s\n  command =%s\n" rule.name (Expr.to_string rule.command)
```

Comment:
I would advise to use the `Format` module that takes a formatter instead of this `to_string` function. See for instance https://github.com/CatalaLang/catala/blob/master/compiler/dcalc/print.ml for an example of how to use `Format`.